jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -285,6 +289,35 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
         }
     }
 
+    private List<String> getPreferredEncodingOrder(HttpServletRequest request)
+    {
+        Enumeration<String> e = request.getHeaders(HttpHeader.ACCEPT_ENCODING.asString());
+        if (e==null || !e.hasMoreElements())
+            return emptyList();
+
+        String key=e.nextElement();
+        while (e.hasMoreElements())
+            key+='\n'+e.nextElement();
```

Comment:
I was aiming at optimizing at the common case of only one header. And creating a StringBuilder for that would have been slower. I now rewrote the code to only use getHeader instead of getHeaders to match what jetty (and all other web servers) have always done.